### ISSUE TITLE
loss function的优化

### ISSUE THREAD

**Issue by dusmart** [zhengwh/captcha-tensorflow#6]
如果最终要预测的是四个字符，把loss定义为四部分交叉熵loss的平方和（每部分label是一个one-hot向量），效果会显著提升，现在这样直接把输出结果和一个four-hot向量做交叉熵，意义不大

**Comment by dusmart** [comment]
采用如上loss，一个1080的卡，保证半个小时内就能收敛到一个很好的结果(>90%)

**Comment by taotaolin** [comment]
> 如果最终要预测的是四个字符，把loss定义为四部分交叉熵loss的平方和（每部分label是一个one-hot向量），效果会显著提升，现在这样直接把输出结果和一个four-hot向量做交叉熵，意义不大

大佬请教一下就是最后输出4个one hot向量作为预测结果是么？

**Comment by dusmart** [comment]
> > 如果最终要预测的是四个字符，把loss定义为四部分交叉熵loss的平方和（每部分label是一个one-hot向量），效果会显著提升，现在这样直接把输出结果和一个four-hot向量做交叉熵，意义不大
> 
> 大佬请教一下就是最后输出4个one hot向量作为预测结果是么？

对的，每一个one hot预测一个字符
